docs/adding_packages/README.md

```diff
@@ -3,180 +3,131 @@
+* Options should [follow these recommendations](conanfile_attributes.md#options) as well as match the default value used by the upstream project.
+* [Package information](build_and_package.md), libraries, components should match as well. This includes exposing supported build system names.
 
-Inside the `conanfile.py` recipe, this data is available in a `self.conan_data` attribute that can be used as follows:
+Where dependencies are involved, there's no shortcuts, inspect the upstream's build scripts for how they usually consume them. Pick the Conan
+generator that matches. The most common example is CMake's `find_package` that can be satisfied by Conan's
+[`CMakeDeps`](https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmakedeps.html) generator. There are a few
+things to be cautious about, many projects like to "vendor" other projects within them. This can be files checked into the repository or
+downloaded during the build process. These should be removed, see the [Dependencies section](dependencies.md#handling-internal-dependencies)
+for more information.
 
-```py
-def export_sources(self):
-    export_conandata_patches(self)
+### How to provide a good recipe
 
-def source(self):
-    files.get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+Take a look at existing [recipes](https://github.com/conan-io/conan-center-index/tree/master/recipes) available in ConanCenterIndex which can be
+used as good examples, you can use them as the base for your recipe. The GitHub search is very good for matching code snippets, you can see if,
+how or when a function is used in other recipes.
 
-def build(self):
-    files.apply_conandata_patches(self)
-    [...]
-```
+> **Note**: Conan features change over time and our best practices evolve so some minor details may be out of date due to the vast number of recipes.
 
-### The _recipe folder_: `conanfile.py`
+More often then not, ConanCenter recipes are built in more configuration then the upstream project. This means some edge cases need minor tweaks.
+We **strongly encourage** everyone to contribute back to the upstream project. This reduce the burden of re-applying patches and overall makes the
+the code more accessible.
 
-The main files in this repository are the `conanfile.py` ones that contain the logic to build the libraries from sources for all the configurations,
-as we said before there can be one single recipe suitable for all the versions inside the `all` folder, or there can be several recipes targeting
-different versions in different folders. For maintenance reasons, we prefer to have only one recipe, but sometimes the extra effort doesn't worth
-it and it makes sense to split and duplicate it, there is no common rule for it.
+Read the docs! The [FAQs](../faqs.md) are a great place to find short answers.
+The documents in this folder are written to explain each folder, file, method and attribute and specific conventions.
```

Comment:
```suggestion
The documents in this folder are written to explain each folder, file, method, and attribute and specific conventions.
```
counting with double add is confusing, maybe use Oxford comma or rewrite

Reply:
```suggestion
The documents in this folder are written to explain each folder, file, method, and attribute.
```